### PR TITLE
C100 Rebuild - Backend API changes

### PR DESCRIPTION
**Pull request by channa-patil**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -22,5 +22,22 @@
...
+  <suppress>
+    <notes><![CDATA[
+      jackson-databind-2.13.3.jar ]]></notes>
+    <cve>CVE-2022-42004</cve>
+  </suppress>
+  <suppress>
+    <notes><![CDATA[
+      jackson-databind-2.13.3.jar ]]></notes>
+    <cve>CVE-2022-38752</cve>
+  </suppress>
```
Comment (AlokDatta) [inline config/owasp/suppressions.xml:42]: We should not have any changes in here.

File: charts/prl-cos/values.preview.template.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ java:
     IDAM_CLIENT_REDIRECT_URI: https://prl-cos-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal/oauth2/callback
     IDAM_API_URL: https://idam-api.{{ .Values.global.environment }}.platform.hmcts.net
     IDAM_S2S_AUTH_URL: http://rpe-service-auth-provider-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
-    CORE_CASE_DATA_API_URL: http://ccd-data-store-api-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal
+    CORE_CASE_DATA_API_URL: http://prl-ccd-definitions-pr-543-ccd-data-store-api
```
Comment (AlokDatta) [inline charts/prl-cos/values.preview.template.yaml:33]: please revert this change before you merge

File: build.gradle [hunk 3]
```diff
@@ -542,6 +541,7 @@ def sonarExclusions = [
   '**uk/gov/hmcts/reform/prl/services/**/*',
   '**uk/gov/hmcts/reform/prl/controllers/citizen/*',
   '**uk/gov/hmcts/reform/prl/services/citizen/*',
+  '**uk/gov/hmcts/reform/prl/services/ManageOrderService.java',
   '**uk/gov/hmcts/reform/prl/repositories/**/*'
```
Comment (AlokDatta) [inline build.gradle:545]: we are trying to address this in 604

File: charts/prl-cos/values.preview.template.yaml (hunk 2 above)
Comment (channa-patil) [reply charts/prl-cos/values.preview.template.yaml:33]: Ack

File: src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseControllerFunctionalTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                        + "YXQyLmludGVybmFsOjg0NDMvb3BlbmFtL29hdXRoMi9yZWFsbXMvcm9vdC9yZWFsbXMvaG1jdHMiLCJ0b2tlb"
+                        + "k5hbWUiOiJhY2Nlc3NfdG9rZW4iLCJ0b2tlbl90eXBlIjoiQmVhcmVyIiwiYXV0aEdyYW50SWQiOiJ1ZnFvMTVXe"
+                        + "VpvZmkwbXUwdFhPUEtTanRkS0UiLCJhdWQiOiJjbWNfY2l0aXplbiIsIm5iZiI6MTY1ODg0ODM0MSwiZ3JhbnRfdHl"
+                        + "wZSI6ImF1dGhvcml6YXRpb25fY29kZSIsInNjb3BlIjpbIm9wZW5pZCIsInByb2ZpbGUiLCJyb2xlcyJdLCJhdXRoX3"
+                        + "RpbWUiOjE2NTg4NDgzNDEsInJlYWxtIjoiL2htY3RzIiwiZXhwIjoxNjU4ODc3MTQxLCJpYXQiOjE2NTg4NDgzNDEsIm"
+                        + "V4cGlyZXNfaW4iOjI4ODAwLCJqdGkiOiJZT215NHQ3NWhCXzlpaC12Vkg5b3ZkLXhaUEEifQ.xQL_vGcoxUMPy88nzm"
+                        + "piY5Hbjp7QmHXSLt-Oxi-yFbQ-d3rUw7MlQtp8NrCetr-OgipgJguL7Sx_E1aiOcOEEjuU_axvS8eVdb143gnwO_vy7e"
+                        + "MBfewrT9PDR7ehU_7gVzCqLlTegPp4SMaW6ffANvQXaVBUOrABC8xFpE_RD1kcyo-nOslKH-MEW5rnXhQZjeRhCA2CUwfj"
+                        + "DP5Qkc7oT5ZO3wIq2EngnDPl9WxhJiKltzavVA64AcAsdy5zdjPRJ-kMPvpZtkC1oUonXslddxeHSQQHaQYxlmiUr1M3f_DL"
```
Comment (AlokDatta) [inline src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseControllerFunctionalTest.java:54]: Do we need this? session token?

File: config/owasp/suppressions.xml (hunk 1 above)
Comment (channa-patil) [reply config/owasp/suppressions.xml:42]: Ok, removed these suppressions.

File: charts/prl-cos/values.preview.template.yaml (hunk 2 above)
Comment (channa-patil) [reply charts/prl-cos/values.preview.template.yaml:33]: Reverted back dev ccd definitions.

File: build.gradle (hunk 3 above)
Comment (channa-patil) [reply build.gradle:545]: You can rebased with master once this PR is merged please.
If we remove these our build will fail & we won't be able to merge PR

File: src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseControllerFunctionalTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    private final String targetInstance =
+            StringUtils.defaultIfBlank(
+                    System.getenv("TEST_URL"),
+                    "http://localhost:4044"
+            );
+
+    private final RequestSpecification request = RestAssured.given().relaxedHTTPSValidation().baseUri(targetInstance);
+
+    @Ignore
```
[the remark this answers is not in the document]
Comment (channa-patil) [reply src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseControllerFunctionalTest.java:37]: Corrected to @Test for this test case

File: src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseControllerFunctionalTest.java (hunk 4 above)
Comment (channa-patil) [reply src/functionalTest/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseControllerFunctionalTest.java:54]: Removed hardcoded tokens & added 
`idamTokenGenerator.generateIdamTokenForCitizen()`

File: build.gradle [hunk 6]
```diff
@@ -95,7 +95,9 @@ allprojects {
   }
 
   repositories {
+    mavenLocal()
```
Comment (channa-patil) [inline build.gradle:98]: @yogendraU Please remove these mavenLocal & jcenter, pushed by mistake I think

**Review by yogendraU — APPROVED** [review]

**Pull request merged**